File: src/main/java/me/zeroeightsix/kami/event/KamiEvent.java
package me.zeroeightsix.kami.event;

import me.zero.alpine.type.Cancellable;
import me.zeroeightsix.kami.util.Wrapper;

/**
 * Created by 086 on 16/11/2017.
 */
public class KamiEvent extends Cancellable {

    private Era era = Era.PRE;
    private final float partialTicks;

    public KamiEvent() {
        partialTicks = Wrapper.getMinecraft().getRenderPartialTicks();
    }

    public Era getEra() {
        return era;
    }

    public float getPartialTicks() {
        return partialTicks;
    }

    public enum Era {
        PRE, PERI, POST
    }

}

File: src/main/java/me/zeroeightsix/kami/event/KamiEvent.kt
package me.zeroeightsix.kami.event

import me.zeroeightsix.kami.util.Wrapper

/**
 * Created by 086 on 16/11/2017.
 * Updated by Xiaro on 18/08/20
 */
open class KamiEvent {
    protected val mc = Wrapper.minecraft
    open val partialTicks: Float = mc.renderPartialTicks
    var era: Era = Era.PRE
    var isCancelled = false

    fun cancel() {
        isCancelled = true
    }

    enum class Era {
        PRE, PERI, POST
    }
}